linalg/sparsemat.cpp

```diff
@@ -23,55 +23,75 @@
 #include <limits>
 #include <cstring>
 
+#if defined(MFEM_USE_CUDA)
+#define cu_or_hip(stub) cu##stub
```

Comment:
Having these `*_or_*` here does not seem the right place: it might also be useful for other related PR.

Reply:
I asked Tom to move these here -- currently they are not needed anywhere else.